common/buildcraft/core/client/RenderTickListener.java

```diff
@@ -106,56 +110,33 @@ public static void renderOverlay(RenderGameOverlayEvent.Text event) {
         if (mouseOver == null) {
             return;
         }
-        boolean both = BCCoreConfig.useLocalServerOnClient;
-
-        IDebuggable client = getDebuggableObject(mouseOver);
-        IDebuggable server = both ? getServer(client) : null;
-
-        if (client == null) return;
-        EnumFacing side = mouseOver.sideHit;
-        if (server == null) {
-            client.getDebugInfo(left, right, side);
-        } else {
-            List<String> serverLeft = new ArrayList<>();
-            List<String> serverRight = new ArrayList<>();
 
+        getDebuggableObject(mouseOver).ifPresent(client -> {
+            EnumFacing side = mouseOver.sideHit;
+            MessageManager.sendToServer(new MessageDebuggableRequest(client.getPos(), side));
```

Comment:
This sends a message to the server, every frame. And on the server it gets debugging information, and sends it back -- note that some clients might have upwards of 120 fps, which is over 6 times the absolute maximum that the information might change (20 times a second).